static/js/stream_edit.js

```diff
@@ -489,6 +490,24 @@ export function initialize() {
         }
 
         stream_settings_ui.sub_or_unsub(sub);
+        const topic = narrow_state.topic();
+        // Refresh the view only if the stream is unsubscribed before
+        // and subscribed now
+        if (!sub.subscribed) {
+            if (topic === undefined) {
+                narrow.activate([{operator: "stream", operand: sub.name}], {
+                    trigger: "subscribe button",
+                });
+            } else {
+                narrow.activate(
+                    [
+                        {operator: "stream", operand: sub.name},
+                        {operator: "topic", operand: topic},
+                    ],
+                    {trigger: "subscribe button"},
+                );
+            }
```

Comment:
This code is wrong; it seems to not be conditional on your actually looking at a view for the stream the user was subscribed to.